dvc/stage/cache.py

```diff
@@ -177,11 +178,19 @@ def _transfer(func, from_remote, to_remote):
 
     def push(self, remote):
         remote = self.repo.cloud.get_remote(remote)
-        return self._transfer(remote.upload, self.repo.cache.local, remote)
+        return self._transfer(
+            _log_exceptions(remote.tree.upload, "upload"),
```

Comment:
Added this without moving it to utils or anything. This is a temporary hack, our tree/remote/cache separation will soon allow us to properly handle this. Ideally, we would want to make remote handle run-cache by itself, since it is really not that much different from regular cache, just that the location is unusual.

Reply:
Thanks for the help.